Sources/CryptorRSA/CryptorRSA.swift

```diff
@@ -446,6 +451,185 @@ public class CryptorRSA {
 			#endif
 		}
 		
+        ///
+        /// Encrypt the data using AES GCM SHA1 for cross platform support.
+        #if os(Linux)
+        func encryptedGCM(with key: PublicKey) throws -> EncryptedData? {
+            // Allocate memory for encryption
+            let rsaEncryptCtx = EVP_CIPHER_CTX_new_wrapper()
+            EVP_CIPHER_CTX_init_wrapper(rsaEncryptCtx)
+            let aeskey = UnsafeMutablePointer<UInt8>.allocate(capacity: 16)
+            let encryptedKey = UnsafeMutablePointer<UInt8>.allocate(capacity: 128)
+            let tag = UnsafeMutablePointer<UInt8>.allocate(capacity: 16)
+            let encrypted = UnsafeMutablePointer<UInt8>.allocate(capacity: data.count + 16)
+            defer {
+                // On completion deallocate the memory
+                EVP_CIPHER_CTX_reset_wrapper(rsaEncryptCtx)
+                EVP_CIPHER_CTX_free_wrapper(rsaEncryptCtx)
+                
+                #if swift(>=4.1)
+                aeskey.deallocate()
+                encryptedKey.deallocate()
+                tag.deallocate()
+                encrypted.deallocate()
+                #else
+                aeskey.deallocate(capacity: 16)
+                encryptedKey.deallocate(capacity: 128)
+                tag.deallocate(capacity: 16)
+                encrypted.deallocate(capacity: data.count + 16)
+                #endif
+            }
+            
+            var processedLength: Int32 = 0
+            var encLength: Int32 = 0
+            // Apple use a 16 byte all 0 IV. This is allowed since a random key is generated for each encryption.
+            let iv = [UInt8](repeating: 0, count: 16)
+            
+            // TODO: change EVP_aes_128_gcm to EVP_aes_256_gcm for rsaKey > 4096 bits to match apple.
```

Comment:
Are we doing this in a separate PR?

Reply:
I have added in the code to choose algorithm based on key size so that this supports RSA keys >= 4096 bytes. I have also added a test to cover this case.